### ISSUE TITLE
Can't open private window if user hasn't allowed the "Run in Private Windows" permission

### ISSUE THREAD

**Issue by 0xRustlang** [cpeterso/open-page-in-private-window#8]
Hi

the addon won't work in Firefox 67.0.1
How i can debug it?
is there any about:config pref that may prevent addon working?

Thanks

**Comment by cpeterso** [comment]
Is there a particular website that is having problems? Or does the add-on always fail? What OS are you using?

You might see some helpful error messages in Firefox's Browser Console (Ctrl+Shift+J on Windows and Linux or Cmd+Shift+J on macOS) when you click the toolbar button.

btw, the add-on doesn't work on Firefox's about: pages like about:config or about:preferences. This is a limitation of Firefox's WebExtension API.

**Comment by 0xRustlang** [comment]
I use Windows 10.
Add-on don't do anything when I press its icon or click on it in context menu.

Thanks
I will check it.

Yes, I know, I changed a lot of prefs in about:config and wanted to know if may be one of them caused this.
I will try on a clean profile and report back tomorrow.
Thanks

**Comment by 0xRustlang** [comment]
Hi again
Sorry for being late.

I tested also on a clean Firefox profile but it didn't worked.
May this be because i use portable version of Firefox?

**Comment by 0xRustlang** [comment]
Oh, Sorry
It was my mistake.
I hadn't allowed addon to be run in private windows ! (new Firefox 67+ permission if i remember correctly)
I allowed it now and worked flawlessly.
Thank you very much :)

**Comment by cpeterso** [comment]
> I hadn't allowed addon to be run in private windows ! (new Firefox 67+ permission if i remember correctly)

Oh! That's interesting. I would not think that the add-on would need to be able to run in private windows to open a new private window, but you're right.

I will add a note about needing the "Run in Private Windows" permission to the extension description. I'm reopening this issue as a reminder to fix this.

**Comment by solipsist** [comment]
I had the same issue and allowing it in private windows helped. I figured the add-on wasn't compatible with recent versions of Fx and considered submitting a patch. Good thing it was such a simple fix.

Please update the description on the addon's page. It's such a useful extension in this era with websites putting tracking cookies wherever you go. Every time I get one of those consent modals I open the page in private mode so I can accept "required" cookies. Then I close the private window and throw away the session. Problem is, pages I've visited don't show up in my history which is a shame since I may want to revisit them in the future.

**Comment by 0xRustlang** [comment]
> Every time I get one of those consent modals I open the page in private mode so I can accept "required" cookies. Then I close the private window and throw away the session. Problem is, pages I've visited don't show up in my history which is a shame since I may want to revisit them in the future.

If you liked use webannoyance (https://github.com/yourduskquibbles/webannoyances ) filterlist and also there are several cookie dialog blocking filterlists on filterlists.com

They are not bulletproof but reduce this annoyances.

**Comment by cpeterso** [comment]
> I had the same issue and allowing it in private windows helped. I figured the add-on wasn't compatible with recent versions of Fx and considered submitting a patch. Good thing it was such a simple fix.

Thanks. I'll take a look. Someone told me how I could have the extension itself check at runtime whether it has the necessary "Run in Private Windows" permission and then show a user warning. But I haven't gotten around to it. 😊

**Comment by cpeterso** [comment]
I added a warning about needing the "Run in Private Windows" permission to the [extension's addons.mozilla.org page](https://addons.mozilla.org/en-US/firefox/addon/open-page-in-private-window/).

I also published a new version (1.2.1) that detects at runtime if the extension is missing the "Run in Private Windows" permission and will show the user a helpful message. That should allow existing users to know why the extension stopped working and how to fix it.

Thanks again for reporting this bug! 😃
